agent/qrexec-client-vm.c

```diff
@@ -250,14 +258,16 @@ int main(int argc, char **argv)
 
     trigger_fd = connect_unix_socket(agent_trigger_path);
 
-    hdr.type = MSG_TRIGGER_SERVICE3;
+    hdr.type = MSG_TRIGGER_SERVICE4;
     hdr.len = sizeof(params) + service_name_len;
 
     memset(&params, 0, sizeof(params));
 
+    if (source_qube!=NULL)
+        strncpy(params.source_domain, source_qube, sizeof(source_qube) - 1);
```

Comment:
```suggestion
        strncpy(params.source_domain, source_qube, sizeof(params.source_domain) - 1);
```
First of all, when copying string, limit copy size to the _target_ space, otherwise any limitation wouldn't prevent overflow. But then, `sizeof(char*)` is 8, so also not what you want.